book/src/development/basics.md

```diff
@@ -0,0 +1,172 @@
+# Basics for hacking on Clippy
```

Comment:
I didn't review the sub-chapters of the development chapter, but right before merging this, we should re-copy those files, so those are up-to-date.  Same for Infrastructure.